File: Spring/next-board/src/main/java/com/example/nextboard/web/member/MemberService.java
package com.example.nextboard.web.member;

import com.example.nextboard.domain.member.entity.Member;
import com.example.nextboard.domain.member.repository.MemberRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class MemberService {

    private final MemberRepository memberRepository;

    //회원 생성 로직
    @Transactional
    public void createMember(Member member) {
        memberRepository.save(member);
    }

    //회원 조회 로직(id)
    public Member findMember(Long memberId) {
        return memberRepository.findById(memberId).orElseThrow(RuntimeException::new);  //예외 처리 필요
    }

    //회원 조회 로직(name)
    public Member findMember(String name) {
        return memberRepository.findByName(name).orElseThrow(RuntimeException::new);
    }
}

File: Spring/kotlin-practice/src/main/kotlin/com/practice/kotlinpractice/service/MemberService.kt
package com.practice.kotlinpractice.service

import com.practice.kotlinpractice.domain.Member
import com.practice.kotlinpractice.domain.MemberRepository
import com.practice.kotlinpractice.domain.Role
import com.practice.kotlinpractice.domain.Role.ROLE_USER
import com.practice.kotlinpractice.dto.MemberSignUpDto
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class MemberService(
    private val memberRepository: MemberRepository,
    private val passwordEncoder: PasswordEncoder
) {

    @Transactional
    fun signUp(memberSignUpDto: MemberSignUpDto) {
        memberRepository.findByEmail(memberSignUpDto.email)
            ?.let { throw Exception("이미 존재하는 이메일입니다.") }
        memberRepository.findByNickname(memberSignUpDto.nickname)
            ?.let { throw Exception("이미 존재하는 닉네임입니다.") }

        val member = Member(
            email = memberSignUpDto.email,
            password = memberSignUpDto.password,
            nickname = memberSignUpDto.nickname,
            age = memberSignUpDto.age,
            city = memberSignUpDto.city,
            role = ROLE_USER
        )

        member.passwordEncode(passwordEncoder)
        memberRepository.save(member)
    }
}